cmd/client/main.go

```diff
@@ -0,0 +1,304 @@
+	fmt.Println("  save-text            -text=<text>  -meta=<meta>")
+	fmt.Println("  save-card            -number=<card_number> -exp=<expiration_date> -cvv=<cvv> -holder=<card_holder_name> -meta=<meta>")
+	fmt.Println("  save-file            -file=<file_path>  -meta=<meta>")
+	fmt.Println("  sync")
+}
+
+func main() {
+	serverURL := flag.String("server", "http://127.0.0.1:8080", "Server URL")
+	grpcServerURL := flag.String("grpc-server", "127.0.0.1:50051", "grpc server URL")
+	dbPath := flag.String("db", "data/client.db", "Path to local BoltDB file")
+	flag.Parse()
+	if flag.NArg() < 1 {
+		printUsage()
+		os.Exit(1)
+	}
+	command := flag.Arg(0)
+	// Создаем базовый контекст с таймаутом.
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	// Открываем локальное хранилище BoltDB.
+	localDB, err := bbolt.OpenLocalStorage(*dbPath)
+	if err != nil {
+		fmt.Println("Error opening local database:", err)
+		os.Exit(1)
+	}
+	defer localDB.Close()
+
+	cli := client.NewClient(*serverURL, *grpcServerURL, session.NewSession(), localDB)
+
+	switch command {
+	case "register":
+		register(ctx, cli, flag.Args()[1:])
+	case "login":
+		login(ctx, cli, flag.Args()[1:])
+	default:
+		// Для остальных проверяем наличие сессии.
+		if cli.Session.GetUserID() == "" {
+			fmt.Println("No session found, please login first.")
+			os.Exit(1)
+		}
+	}
+
+	switch command {
+	case "get":
+		getItems(ctx, cli, flag.Args()[1:])
+	case "save-credential":
+		saveCredentials(ctx, cli, flag.Args()[1:])
+	case "save-text":
+		saveText(ctx, cli, flag.Args()[1:])
+	case "save-card":
+		saveCard(ctx, cli, flag.Args()[1:])
+	case "save-file":
+		saveFile(ctx, cli, flag.Args()[1:])
+	case "sync":
+		sync(ctx, cli, flag.Args()[1:])
+	case "delete":
+		delete(ctx, cli, flag.Args()[1:])
+	default:
+		if command != "register" && command != "login" {
+			fmt.Println("Unknown command:", command)
+			printUsage()
+			os.Exit(1)
+		}
+	}
+}
+
+func register(ctx context.Context, cli *client.Client, args []string) {
+	cmd := flag.NewFlagSet("register", flag.ExitOnError)
+	username := cmd.String("username", "", "Username")
+	password := cmd.String("password", "", "Password")
+	if err := cmd.Parse(args); err != nil {
+		fmt.Println("Failed to parse arguments")
+		os.Exit(1)
+	}
+	if *username == "" || *password == "" {
+		fmt.Println("username and password must be provided")
+		os.Exit(1)
+	}
+	dto := client.RegisterDTO{
+		Username: *username,
+		Password: *password,
+	}
+	if err := cli.Register(ctx, dto); err != nil {
+		fmt.Println("Registration error:", err)
+		os.Exit(1)
+	}
+	fmt.Println("Registration successful")
+}
+
+func login(ctx context.Context, cli *client.Client, args []string) {
+	cmd := flag.NewFlagSet("login", flag.ExitOnError)
+	username := cmd.String("username", "", "Username")
+	password := cmd.String("password", "", "Password")
+	if err := cmd.Parse(args); err != nil {
+		fmt.Println("Failed to parse arguments")
+		os.Exit(1)
```

Comment:
```suggestion
		log.Fatal("Failed to parse arguments")
```
Там внутри как раз две этих функции.